src/workload.rs

```diff
@@ -639,10 +620,14 @@ impl WorkloadStore {
                 }
             }
         }
-        if Some(&w.node) == self.local_node.as_ref()
-            || (Some(&w.name) == self.local_pod.as_ref()
-                && Some(&w.namespace) == self.local_namespace.as_ref())
-        {
+        let should_fetch: bool = match self.proxy_mode {
+            ProxyMode::Sidecar => {
+                Some(&w.name) == self.local_pod.as_ref()
+                    && Some(&w.namespace) == self.local_namespace.as_ref()
```

Comment:
Curious if in sidecar node, whose certificate would ztunnel use?

Reply:
> Curious if in sidecar node, whose certificate would ztunnel use?

@hzxuzhonghu Of course it is in the pod where the ztunnel is located (ztunnel is not an independent pod, but a sidecar container in the pod), the pod name and namespace comes from env `POD_NAME` and `POD_NAMESPACE` using downward api, just like the sidecar mode.

Reply:
Then how could ztunnel be injected to workload container? And also curious why not sidecar in this case.

Reply:
> Then how could ztunnel be injected to workload container? And also curious why not sidecar mode in this case.

@hzxuzhonghu The core reason is that we want to integrate ztunnel in the serverless environment. In our serverless environment, each pod occupies a lightweight virtual machine exclusively, and ztunnel cannot be deployed by daemonset.
We want to integrate ztunnel into serverless pods in a sidecar-like way and enable it by default at scale.
The reason for not using sidecar directly is that in large-scale scenarios, resource overhead and performance loss 
 of envoy cannot be ignored, which is not conducive to large-scale use.

Reply:
> Then how could ztunnel be injected to workload container? 

@hzxuzhonghu In our case, ztunnel is a hidden sidecar container, injected by the serverless agent in the lightweight virtual machine at runtime, user can not see the ztunnel container in pod.

Reply:
I believe that other similar serverless environments will also use the ability of `PROXY_MODE`=sidecar to integrate with ambient mesh.

Reply:
Can you explain how L25 equals ` Some(&w.name) == self.local_pod.as_ref()`

Reply:
It's compares the pod name, just like the node name comparison when `PROXY_MODE`=node:

```rust
Some(&w.node) == self.local_node.as_ref()
```

Reply:
As you say workload and ztunel are two pods And w.name actually is not pod name just pod IP.

Reply:
No, I'm talking about ztunnel and workload are the same pod. 

And w.name actually the pod name not pod IP, check this out: https://github.com/istio/ztunnel/blob/ac83b1c51e0f7d57162fad858acc76b2bb0546b4/proto/workload.proto#L22

Reply:
I guess the desc is not right, from istiod , I remember it is pod IP address

Reply:
there is Resource.Name and the "name" field in workload proto they are distinct things